core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -780,7 +780,9 @@ private void handleRequestOrResponseComplete() {
                         // Stop receiving new requests.
                         handledLastRequest = true;
                         if (unfinishedRequests.isEmpty()) {
-                            ctx.writeAndFlush(Unpooled.EMPTY_BUFFER).addListener(CLOSE);
+                            ctx.executor().schedule(() ->
```

Comment:
`ServiceRequestContext#executor` isn't necessarily the same as `Channel#eventLoop`

https://github.com/line/armeria/blob/1d268aa278524997ad5c1cc8f66661cfabbb6dd1/core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java#L375-L387

We probably want to close from the channel's event loop (otherwise, an exception will be raised)
```suggestion
                            ctx.channel().eventLoop().schedule(() ->
```

Reply:
Oh I missed it. Thanks for letting me know.
I fixed it to use the channel's event loop.